SUSE AI

tbazant opened the pull request:
### Description

This update introduces a brand new documentation for the SUSE AI product

rtamalin commented on tasks/NVIDIA-GPU-driver-installation.xml:
```diff
@@ -0,0 +1,413 @@
+        <para>
+          You can then install the driver and
+          <package>nvidia-compute-utils</package> for additional utilities that
+          are worth for testing purposes.
```
`worth` doesn't make sense in this context...

```suggestion
          are worthwhile having for testing purposes.
```
or possibly
```suggestion
          are useful for testing purposes.
```

rtamalin commented on tasks/NVIDIA-GPU-driver-installation.xml:
```diff
@@ -0,0 +1,413 @@
+  <section xml:id="nvidia-gpu-validation">
+    <title>Validation of the driver installation</title>
+    <para>
+      The <command>nvidia-smi</command> command has verified that, at the host
```
The first sentence here feels like it is missing something. I think the following change would resolve that:

```suggestion
      Running the <command>nvidia-smi</command> command has verified that, at the host
```

rtamalin commented on tasks/NVIDIA-GPU-driver-installation.xml:
```diff
@@ -0,0 +1,413 @@
+<screen>&prompt.root;<command>nvidia-ctk cdi generate --output=/etc/cdi/nvidia.yaml</command></screen>
+        <para>
+          This ensures that any container started on the machine can employ
+          discovered &nvidia; GPU devices. You can then run a &podman;-based
```
From `You can then run a &podman;-based` onwards feels like it should be part of the next step's initial text, rather than buried at the end of this step...

jeremy-moffitt commented on articles/NVIDIA-Operator.asm.xml:
```diff
@@ -0,0 +1,116 @@
+            <term>WHY?</term>
+            <listitem>
+              <para>
+                To learn how to automate the installation, configuration and
```
We may need to be cautious about the word "automate" here...  maybe "streamline" or "simplify"... I dont think we've successfully developed an automated solution... cc @victorstevansuse

jeremy-moffitt commented on concepts/AI-preventing-hallucinations.xml:
```diff
@@ -0,0 +1,312 @@
+ xmlns:xlink="http://www.w3.org/1999/xlink"
+ xmlns:trans="http://docbook.org/ns/transclusion">
+  <info>
+    <title>Preventing AI hallucinations</title>
```
will ping zack.feldstein@suse.com  on this section

jeremy-moffitt commented on images/ai-schema.png:
We need to get the marketing images for this rather than my janky lucid charts

jeremy-moffitt commented on concepts/NVIDIA-Operator.xml:
```diff
@@ -0,0 +1,129 @@
+      </listitem>
+      <listitem>
+        <para>
+          <emphasis role="bold">Scheduling GPU workloads.</emphasis> Once the
```
we owe you some detailed content on this

jeremy-moffitt reviewed (commented):
Looks like a solid start! We definitely need to replace my janky lucid chart, and I will ask the team to provide some additional reviews and content asap.

jeremy-moffitt commented:
![suse_ai_stack](https://github.com/user-attachments/assets/523890a4-0bc3-49ec-90dc-bdfeed75e434)
consider using this in place of the ai-schema.png file

jeremy-moffitt commented:
![suse_ai_suite](https://github.com/user-attachments/assets/df7086f0-fca6-41db-8384-7892ee8fa208)
this one has the updated branding for NeuVector (aka SUSE Security)

jeremy-moffitt commented:
![suse_ai_suite](https://github.com/user-attachments/assets/fb34fba9-f21b-4c12-baac-a331387048b6)
Third time is the charm... deleted previous two images... I think this one has the updated names and covers the full suite

jeremy-moffitt commented:
![suse_ai_full](https://github.com/user-attachments/assets/6b62d0af-96fa-4098-952f-a9dbef64e6cd)
Third time is the charm... deleted previous two images... I think this one has the updated names and covers the full suite

jeremy-moffitt commented:
> ![suse_ai_full](https://private-user-images.githubusercontent.com/23247873/377618221-6b62d0af-96fa-4098-952f-a9dbef64e6cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1MTA1ODcsIm5iZiI6MTcyOTUxMDI4NywicGF0aCI6Ii8yMzI0Nzg3My8zNzc2MTgyMjEtNmI2MmQwYWYtOTZmYS00MDk4LTk1MmYtYTlkYmVmNjRlNmNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIxVDExMzEyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcxMjM4ZTk2NjlmZDE0N2I3ZmJhNmFkZTM5NTU4NmVlZjRkNWUyY2I3MTJiMDZkYWFjYzhkMDAwNWM2NmIwMjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.t4iFqkrCCxQL8tmLBrvJPAJ6uIrVNHhqm5BoB2rrDn8) Third time is the charm... deleted previous two images... I think this one has the updated names and covers the full suite

it occurs to me that I deleted the comment indicating what this should be used for... this should replace the ai-schema picture

rewenila commented:
@tbazant I took some new screenshots of Open WebUI with the SUSE branding, tried to replicate the content of yours as best as I could. Please let me know if I missed something or if you need any additional screenshot.
[openwebui_screenshots.zip](https://github.com/user-attachments/files/17462512/openwebui_screenshots.zip)

tbazant commented:
> @tbazant I took some new screenshots of Open WebUI with the SUSE branding, tried to replicate the content of yours as best as I could. Please let me know if I missed something or if you need any additional screenshot. [openwebui_screenshots.zip](https://github.com/user-attachments/files/17462512/openwebui_screenshots.zip)

thanks a lot, imgs are updated now (not synced yet)

tbazant commented:
> > ![suse_ai_full](https://private-user-images.githubusercontent.com/23247873/377618221-6b62d0af-96fa-4098-952f-a9dbef64e6cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1MTA1ODcsIm5iZiI6MTcyOTUxMDI4NywicGF0aCI6Ii8yMzI0Nzg3My8zNzc2MTgyMjEtNmI2MmQwYWYtOTZmYS00MDk4LTk1MmYtYTlkYmVmNjRlNmNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIxVDExMzEyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcxMjM4ZTk2NjlmZDE0N2I3ZmJhNmFkZTM5NTU4NmVlZjRkNWUyY2I3MTJiMDZkYWFjYzhkMDAwNWM2NmIwMjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.t4iFqkrCCxQL8tmLBrvJPAJ6uIrVNHhqm5BoB2rrDn8) Third time is the charm... deleted previous two images... I think this one has the updated names and covers the full suite
> 
> it occurs to me that I deleted the comment indicating what this should be used for... this should replace the ai-schema picture

thanks, updated

thbertoldi commented on concepts/AI-introduction.xml:
```diff
@@ -0,0 +1,275 @@
+    </para>
+  </section>
+  <section xml:id="ai-scenarios">
+    <title>What are typical scenarios for using &productnameshort;</title>
```
Suggestion: list one use case associated with "building a knowledge base where users can easily interact with chatbots to ask questions about the company policies, processes, and workflows".

thbertoldi commented on concepts/AI-introduction.xml:
```diff
@@ -0,0 +1,275 @@
+        </listitem>
+        <listitem>
+          <para>
+            <emphasis role="bold">&observability;</emphasis> providing advanced
```
I believe that in the rendered document the "observabilitiy" reference is empty
![image](https://github.com/user-attachments/assets/aefcf0ea-39e9-42d9-9b05-b3ec368468e8)

thbertoldi commented on concepts/AI-introduction.xml:
```diff
@@ -0,0 +1,275 @@
+        </listitem>
+        <listitem>
+          <para>
+            <emphasis role="bold">Knowledge retrieval</emphasis>. The AI
```
This section, I believe, should be more explicit about the use of RAG systems, where models are able to get contextual information from the company, not relying on training data or external information.

thbertoldi commented on concepts/AI-introduction.xml:
```diff
@@ -0,0 +1,275 @@
+      </orderedlist>
+    </section>
+  </section>
+  <section xml:id="ai-benefits">
```
Suggestion: One benefit could be SUSE's quality control on the whole software production chain used by the AI stack.

thbertoldi commented on concepts/openwebui-introduction.xml:
Suggestion: Maybe it would be relevant to introduce the following concepts:
* Workspace
* Knowledge Base
* Creating a new model from a existing one + one or more knowledge bases.

thbertoldi reviewed (commented):
Here are a few suggestions for the documents.

rewenila commented on articles/openwebui.asm.xml:
```diff
@@ -0,0 +1,478 @@
+                </listitem>
+                <listitem>
+                  <para>
+                    Selected tasks require the &rootuser; privileges.
```
I think it was supposed to be admin/administrator privileges here.

tbazant replied:
fixed

rewenila commented on common/generic-entities.ent:
```diff
@@ -200,7 +212,16 @@ use &deng;! -->
-
+<!ENTITY stackstate             "StackState">
+<!ENTITY ollama                 "Ollama">
+<!ENTITY owui                   "OpenWebUI">
```
"OpenWebUI" should be replaced with "Open WebUI".

tbazant replied:
fixed

rewenila reviewed (commented):
I have two suggestions related to the Open WebUI articles.

thbertoldi commented on references/AI-requirements-ollama.xml:
```diff
@@ -0,0 +1,144 @@
+          nodes.</emphasis> &productname; provides a mechanism to assign
+          applications such as &ollama; to specific nodes. For more details on
+          how to assign applications to nodes, refer to
+<!-- TODO tbazant: add link to assigning apps to nodes
```
The document is available at Confluence with the name "Node Assignment for Applications."

tbazant replied:
thanks, will start today :-)

thbertoldi commented on tasks/NVIDIA-Operator-installation.xml:
It may be interesting to place a section describing how to use the NVIDIA GPU resources from an application helm chart perspective. I discussed it briefly in the document "Node Assignment for Applications," but maybe it would be better to place it elsewhere. Basically, in the application helm chart the user needs to declare the need for the GPU resources and enable GPU passthrough. Ollama is a good example https://github.com/SUSE/private-ai-charts/tree/main/charts/ollama

jeremy-moffitt commented on concepts/AI-introduction.xml:
```diff
@@ -0,0 +1,294 @@
+          </listitem>
+        </varlistentry>
+        <varlistentry>
+          <term>&langchain; (https://www.langchain.com/)</term>
```
langchain is not available as a standalone component for the initial release, we should remove it as a line-item for now.

guangyee commented on DC-AI-deployment-intro:
I am not sure if the PR is in-sync with the generated HTML here (https://docserv.suse.de:8085/en-us/suse-ai/1.0/html/AI-deployment-intro/) so I'll just provide general feedback.

1. "SUSE AI Requirements" link (https://documentation.suse.com/suse-ai/1.0/html/AI-requirements/index.html) seem broken.
2. In section "1.2 Cluster preparation", we may want to remove the optional step 3 "Install the NVIDIA container toolkit" The container toolkit will be installed as part of NVIDIA GPU Operator.
3. In section "1.2 Cluster preparation", after step "4. Install RKE2 Kubernetes distribution", we need to a step to install the NVIDIA GPU Operator with "--set driver.enabled=false" flag.
4. We may want to remove any reference to AMD for now as we have not test anything against AMD GPUs.

tbazant replied:
> I am not sure if the PR is in-sync with the generated HTML here (https://docserv.suse.de:8085/en-us/suse-ai/1.0/html/AI-deployment-intro/) so I'll just provide general feedback.
> 
>     1. "SUSE AI Requirements" link (https://documentation.suse.com/suse-ai/1.0/html/AI-requirements/index.html) seem broken.
this is due to the fact that we removed the docs from a production server to a testing server
> 
>     2. In section "1.2 Cluster preparation", we may want to remove the optional step 3 "Install the NVIDIA container toolkit" The container toolkit will be installed as part of NVIDIA GPU Operator.

removed

>     3. In section "1.2 Cluster preparation", after step "4. Install RKE2 Kubernetes distribution", we need to a step to install the NVIDIA GPU Operator with "--set driver.enabled=false" flag.

i guess the step was already there, but i got the feeling from Confluence that we dont need it. i'll readd it

> 
>     4. We may want to remove any reference to AMD for now as we have not test anything against AMD GPUs.

there are no such occurencies

tbazant commented on concepts/AI-introduction.xml:
```diff
@@ -0,0 +1,293 @@
+        <listitem>
+          <para>
+            <emphasis role="bold">&nvoperator;</emphasis> to utilize the
+            a&nvidia; GPU computing power and capabilities for processing
```
```suggestion
            &nvidia; GPU computing power and capabilities for processing
```

tbazant commented on concepts/AI-introduction.xml:
```diff
@@ -0,0 +1,293 @@
+              minimal performance loss.
+            </para>
+          </listitem>
+        </varlistentry>
+        <varlistentry>
+          <term>&langchain; (https://www.langchain.com/)</term>
+          <listitem>
+            <para>
+              A framework for building applications powered by language models.
+              It focuses on chaining of multiple model interactions to create
+              complex workflows.
+            </para>
+          </listitem>
+        </varlistentry>
```
```suggestion
        </varlistentry>
        <!--
        <varlistentry>
          <term>&langchain; (https://www.langchain.com/)</term>
          <listitem>
            <para>
              A framework for building applications powered by language models.
              It focuses on chaining of multiple model interactions to create
              complex workflows.
            </para>
          </listitem>
        </varlistentry> -->
```

tbazant commented on concepts/NVIDIA-Operator.xml:
```diff
@@ -0,0 +1,129 @@
+          available hardware.
+        </para>
+      </listitem>
+      <listitem>
+        <para>
+          <emphasis role="bold">&nvidia; driver installation.</emphasis> The
+          operator ensures that the appropriate &nvidia; drivers are installed
+          on the cluster nodes.
+        </para>
+      </listitem>
```
```suggestion
<!--
      <listitem>
        <para>
          <emphasis role="bold">&nvidia; driver installation.</emphasis> The
          operator ensures that the appropriate &nvidia; drivers are installed
          on the cluster nodes.
        </para>
      </listitem> -->
```

tbazant reviewed (commented):
thanks for your suggestions

dariavladykina commented on references/owui-helmchart.xml:
```diff
@@ -0,0 +1,1161 @@
+          </entry>
+          <entry>
+            <para>
+              A list of &ollama; API endpoints. These can be added in lieu of
```
```suggestion
              A list of &ollama; API endpoints. These can be added instead of
```

dariavladykina commented on references/owui-helmchart.xml:
```diff
@@ -0,0 +1,1161 @@
+          </entry>
+          <entry>
+            <para>
+              Use existingClaim to re-use an existing &owui; PVC instead of
```
```suggestion
              Use existingClaim to reuse an existing &owui; PVC instead of
```

dariavladykina commented on references/owui-helmchart.xml:
```diff
@@ -0,0 +1,1161 @@
+          </entry>
+          <entry>
+            <para>
+              Tolerations for pod assignment.
```
```suggestion
              Tolerations for pod assignment
```

dariavladykina commented on references/owui-helmchart.xml:
```diff
@@ -0,0 +1,1161 @@
+          </entry>
+          <entry>
+            <para>
+              Topology Spread Constraints for pod assignment.
```
```suggestion
              Topology Spread Constraints for pod assignment
```

dariavladykina commented on references/owui-helmchart.xml:
```diff
@@ -0,0 +1,1161 @@
+        <term>Let's Encrypt</term>
+        <listitem>
+          <para>
+            This method also uses <literal>cert-manager</literal>but it is
```
```suggestion
            This method also uses <literal>cert-manager</literal>, but it is
```

dariavladykina commented on references/owui-helmchart.xml:
```diff
@@ -0,0 +1,1161 @@
+            actions&mdash;including request and validation&mdash;to get the
+            Let's Encrypt certificate issued. This configuration uses HTTP
+            validation (HTTP-01) and therefore the load balancer must have a
+            public DNS record and be accessible from the internet.
```
```suggestion
            public DNS record and be accessible from the Internet.
```

dariavladykina commented on references/owui-helmchart.xml:
```diff
@@ -0,0 +1,1161 @@
+          </entry>
+          <entry>
+            <para>
+              This section can be used to pass required environment variables to
```
```suggestion
              This section can be used to pass the required environment variables to
```

dariavladykina reviewed (commented):
Hi Tomas, here are the reviews of all files, hope I didn't miss anything. Please send this whole creation to me once again for a final look. Thanks a lot!

dariavladykina reviewed (commented):
Hi Tomas, here are the reviews of all files, hope I didn't miss anything. Please send this whole creation to me once again for a final look. Thanks a lot!

dariavladykina commented on tasks/AI-gpu-nodes-assigning.xml:
```diff
@@ -0,0 +1,159 @@
+    <abstract>
+      <para>
+        When deploying a containerized application to &kube;, you need to ensure
+        that containers that require GPU resources are run on appropriate worker
```
```suggestion
        that containers requiring GPU resources are run on appropriate worker
```

dariavladykina commented on tasks/AI-gpu-nodes-assigning.xml:
```diff
@@ -0,0 +1,159 @@
+    </section>
+  </section>
+  <section xml:id="ai-gpu-assign-verify">
+    <title>Verify &ollama; GPU assignment</title>
```
```suggestion
    <title>Verifying &ollama; GPU assignment</title>
```

dariavladykina commented on tasks/AI-gpu-nodes-assigning.xml:
```diff
@@ -0,0 +1,159 @@
+      <para>
+        If affected cluster nodes are labeled with a label such as
+        <literal>accelerator=nvidia-gpu</literal>, you can configure the node
+        selector to check for the label. In this case, user the following values
```
```suggestion
        selector to check for the label. In this case, use the following values
```

dariavladykina commented on tasks/AI-gpu-nodes-assigning.xml:
```diff
@@ -0,0 +1,159 @@
+  <section xml:id="ai-gpu-assign-verify">
+    <title>Verify &ollama; GPU assignment</title>
+    <para>
+      If GPU is correctly detected, the &ollama; container logs such event:
```
```suggestion
      If the GPU is correctly detected, the &ollama; container logs this
      event:
```

dariavladykina commented on tasks/AI-requirements-apachepulsar.xml:
```diff
@@ -0,0 +1,60 @@
+    <meta name="maintainer" content="" its:translate="no"/>
+    <abstract>
+      <para>
+        Installation of &apulsar; and its dependencies may require updating the
```
```suggestion
        The installation of &apulsar; and its dependencies may require updating the
```

dariavladykina reviewed (commented):
Additional review of two files. Thanks!
